docs/architecture/specs/marketplace_api/README.md

```diff
@@ -0,0 +1,221 @@
+
+### Assets Management
+
+The Assets in a Nevermined ecosystem are represented via DDOs that are resolved via a Decentralized Identifier (DID). While the DIDs are 
+unique identifiers stored on-chain, the metadata of an asset is represented by a DDO recorded in an external data storage (centralized or decentralized).
+This DDO includes all the additional information describing an asset and the services attached to that asset (access, computation, etc.).
+
+This module of the API allows the management of the assets information. The main functionalities provided are the following:
+
+* Create a DDO of a new asset - `POST /api/v1/metadata/assets/ddo`
+* Get the DDO of a particular asset - `GET /api/v1/metadata/assets/ddo/{did}`
+* Update DDO of an existing asset - `PUT ​/api​/v1​/metadata​/assets​/ddo​/{did}`
+* Delete an existing asset - `DELETE ​/api​/v1​/metadata​/assets​/ddo​/{did}`
+
+The Creation, Update and Deletion of assets is authorized and **only the asset owner and/or the application admin** is allowed to do that action.
+
+For more information about DID and/or DDO's please refer to the [DID SPEC: Decentralized Identifiers](../did/README.md) and 
+[META SPEC: Metadata Ontology](../metadata/README.md).
+
+### Assets Search
+
+This module complements the previous one and provides search functionalities on top of the existing assets metadata. All the actions listed
+here don't require user authorization:
+
+* Get all asset IDs - `GET ​/api​/v1​/metadata​/assets`
+* Get DDO of all assets - `GET ​/api​/v1​/metadata​/assets​/ddo`
+* Get a list of DDOs that match with the given text - `GET ​/api​/v1​/metadata​/assets​/ddo​/query`
+* Get a list of DDOs that match with the executed query - `POST ​/api​/v1​/metadata​/assets​/ddo​/query`
+
+
+### Users Profiles
+
+This module of the API allows the management of user profiles. The main functionalities provided are the following:
+
+* Create a user profile - `POST /api/v1/metadata/profiles`
+* Get the metadata of a user profile - `GET /api/v1/metadata/profiles/{userId}`
+* Update the user profile - `PUT /api/v1/metadata/profiles/{userId}`
+* Disable the user profile - `DELETE /api/v1/metadata/profiles/{userId}`
+
+The Creation, Update and Deletion of profiles is authorized and **only the asset owner and/or the application admin** is allowed to do that action.
+
+
+### Secondary Market
+
+This module of the API allows the management of the Secondary Market transactions. The main functionalities provided are the following:
+
+* Create a service agreement entry - `POST /api/v1/agreement/services`
+* Get the a service agreement entry - `GET /api/v1/agreement/services/{serviceId}`
+* Update a service agreement entry - `PUT /api/v1/agreement/services/{serviceId}`
+* Disable a service agreement entry - `DELETE /api/v1/agreement/services/{serviceId}`
+
+The Creation, Update and Deletion of service agreeemnts is authorized and **only the asset owner** is allowed to do that action.
+
+### NFTs Metadata
```

Comment:
Is this related only to the metadata that marketplaces like opensea use?
Or else there is an overlap between this and the asset metadata

Reply:
The intention is to manage the NFTs metadata that OpenSea and similar require.